google/stackdriver_monitoring/spectator_client.py

```diff
@@ -147,12 +151,30 @@ def scan_by_service(self, service_list, params=None):
     if service_list == ['all']:
       service_list = self.SERVICE_PORT_MAP.keys()
 
-    for service in service_list:
-      port = self.SERVICE_PORT_MAP[service]
+    start = time.time()
+    service_time = {service: 0 for service in service_list}
+    result = {service: None for service in service_list}
+    threads = {}
+
+    def timed_collect(self, service): #, params, service_time, result):
+      now = time.time()
       try:
+        port = self.SERVICE_PORT_MAP[service]
         result[service] = self.collect_metrics(self.__host, port, params=params)
       except IOError as ioex:
         logging.getLogger(__name__).error('%s failed: %s', service, ioex)
+      service_time[service] = int ((time.time() - now) * 1000)
+
+    for service in service_list:
+      threads[service] = threading.Thread(
+          target=timed_collect,
+          args=(self, service))#, params, service_time, result))
+      threads[service].start()
+    for service in service_list:
+      threads[service].join()
+
+    logging.info('Collection times %d (ms): %s',
```

Comment:
what's wrong with the way it is? It's just for reference and better to keep on one line for this use case.